lib/engine/game/g_1849/step/buy_train.rb

```diff
@@ -34,6 +37,34 @@ def can_sell?(entity, bundle)
 
             super
           end
+
+          def buyable_trains(entity)
+            # Cannot buy E-train without E-token
+            trains_to_buy = super
+
+            trains_to_buy = trains_to_buy.reject { |t| t.name == 'E' }
+            trains_to_buy << e_train if @game.electric_dreams? && can_buy_e?(entity)
+            trains_to_buy.uniq
+          end
+
+          def e_train
+            @depot.depot_trains.find { |t| t.name == 'E' }
+          end
+
+          def can_buy_e?(entity)
+            entity.e_token &&
+              e_train.price <= entity.cash
+          end
+
+          def check_for_cheapest_train(train)
+            entity = @game.round.current_operator
+
+            cheapest = @depot.send(@game.electric_dreams? && !entity.e_token ? :min_depot_train_no_e_token : :min_depot_train)
```

Comment:
Why do you need to check `electric_dreams?` Isn't checking for `e_token` sufficient? Or does corporation not have `e_token` attr unless electric dreams is enabled?

Reply:
`send` is a weird way to do this. The more standard way is
```
entity.e_token ? @depot.min_depot_train_no_e_token : @depot.min_depot_train
```